### ISSUE TITLE
Import/export 1.7.3

### ISSUE THREAD

**Issue by colla70** [geopaparazzi/smash#176]
After upgrade to 1.7.3 tapping on 'import' or 'export' leads to a grey empty page

**Comment by colla70** [comment]
I'm on Android.
1) upgrade on Samsung Galaxy A40 Android 11.
2) clean install on Samsung A32  Android 11.
3) uninstalled 1.7.3pre (the one I've tested for measure tool issue) reinstalled 1.7.3 on Lenovo Tab M10 FHD Plus Android 10 same issue (one weird thing is that this morning Play Store let me install 1.7.2 then the upgrade..).
On 1) and 2) I've also noticed that:
a) in 'info' tapping on 'legal info' or 'privacy policy' nothing happens
b) trying to add new local layer I can't reach other folders than /storage/emulated/0/Android/data/eu.hydrologis.smash/files (Warning message: The top level folder has already been reached) and this is another big issue for users without rooted devices...
On Lenovo Tab Android 10 no issues on this two last points.

**Comment by colla70** [comment]
Other clue: all devices have italian version of the app...

**Comment by moovida** [comment]
Thanks @colla70 , that might be the issue. Going to convert one of my devices and check.
The other person that signalled me the issue also has an Italian IOS device.

Fingers crossed.

**Comment by colla70** [comment]
Maybe I've been too verbose translating some strings and that can't be handled by GUI... ;)
Can you also check storage permissions or have I to open another issue?

**Comment by moovida** [comment]
Ah, you are the Italian translator! Thanks a lot man! No, no, I assume it is a bug bounds to some language engine thing.

Regarding the storage, there is no possibility to change this after android 10. Google forces applications that do not have their permission, to work in the folder you mention above. And they do give complete disk control permission only to applications like file browsers and antivirus. So we do not have a chanche in this. The other field mapping apps supper the same fait.

What I do everytime from the last version on, is that I also upload an apk of every release to the github release ( red the last note here: https://github.com/geopaparazzi/smash/releases/tag/version_1.7.3%2B65 ) that has the total disk permission. So if you need to work on the complete disk (as I need to), you will need to manually install that one.

**Comment by moovida** [comment]
Anyways @colla70 , thanks, I am now able to reproduce. 
Looking into it.

**Comment by moovida** [comment]
Ok, this is fixed now. 
I also opened a new branch for the import/export plugins (the ones giving issues due to missing language files) for the translations:
https://hosted.weblate.org/projects/smash/smash-import-and-export-plugins/

Right now I just copied over the old ones from smash, so there might be missing bits and pieces.

**Comment by moovida** [comment]
I will try to clean up the translation file.

**Comment by moovida** [comment]
Ok, cleanup has been done. Translating is now safe. Closing this.

**Comment by mbasa-georepublic** [comment]
@moovida . this issue is because of this: 

```
supportedLocales: SL.delegate.supportedLocales,
```

wherein any new Locale SMASH will have should also be included in the Import-Export lib. Otherwise a red screen error will appear when displaying the import or export screens.

**Comment by mbasa-georepublic** [comment]
What's frustrating is though is that the Locales in Import-Export are actually never used, as in it is possible to leave the locale files blank, since the `SL` locales will always be searched first. And if the `SL` does not contain the requested string, the default EN of `IEL` is only searched since the locale information is not passed to the Import-Export lib.  

I have been stuck with this trying to create a fix but I haven't been successful. 

I even tried reversing the order of the delegates: 

```
localizationsDelegates: [
        IEL.delegate,        
        SL.delegate,        
        GlobalMaterialLocalizations.delegate,
        GlobalWidgetsLocalizations.delegate,
      ],
```

but that ended up with `IEL` properly localized but with `SL` only displaying the default EN strings.

**Comment by moovida** [comment]
H @mbasa-georepublic , you mean that the enw released version 1.7.4 still has this issue for you?
Becasue I fixed this (at least I thought and it worked for me with Italian localization) and released a new 1.7.4 version.
